Nitaicandra/CENPROJECT
# Search backend

## Summary of changes
- Added address validation when creating accounts using SmartyStreets API
- Added search functionality by keyword/phrase
- Search returns services for business located within 10 miles from the logged-in customer

## How to test
IMPORTANT: Use state acronyms instead of full state names when creating accounts. This update does not work with full state names (and the current form in the UI)

1. You will need to delete the mongo container,image and volumes and rebuild them from scratch. Old data with fake addresses may break the application
2. Update the .env file within /server to add the API keys (check the resources channel in discord)
3. Create business account(s) and add services. Make sure you are logged in and sending the token in the authorization header
4. Create a customer account and log in. 
5. In Postman, create a new GET request with the path `http://localhost:3001/api/search?query=plumbing`. You can replace this plumbing keyword with whatever other service you want to search. It also works with substrings (like "plumb" should return plumbing services)
5. Make sure you are sending the customer's token in authorization. 

## WARNING
This functionality uses external API calls. We only have 1,000 total requests for free in Smartly so be mindful of the amount of users you will be creating/testing from now on.

The search endpoint uses the google maps api, which we have 1,000/mo free requests and a $300 credit. So we dont need to worry so much about this one.
